dandiapi/api/models/asset_paths.py

```diff
@@ -4,7 +4,7 @@
 
 
 class AssetPath(models.Model):
-    path = models.CharField(max_length=512)
+    path = models.CharField(max_length=512, db_collation='C')
```

Comment:
I think we should have a comment similar to the one in `test_asset_path_ordering` describing why `db_collation` is set here.